### PR TITLE
Move Pulumi Cloud Console docs to their own section

### PR DESCRIPTION
**Pull request by chrsmith**
This is mostly a refactoring PR, moving the Pulumi Service documentation from 

https://www.pulumi.com/docs/reference/service/
https://www.pulumi.com/docs/reference/cd/
to
https://www.pulumi.com/docs/reference/console/
https://www.pulumi.com/docs/reference/console/continuous-delivery

➡️ [Link to deployed preview](https://deploy-preview-1500--pulumi.netlify.com/docs/console/) ⬅️ 

Part of this is consolidating those two content sections from "Teams & Collaboration" and "Continuous Delivery" under a single "Pulumi Cloud Console" section. Here's a screen cap of the new left nav:

image

Questions:

- I put menu metadata directly inside the pages rather than in `menus.toml`. Is that the right call?
- I could use some help help in how we want to organize the docs. Not sure where to put our documentation for Webhooks. Having a nav item for "Features" seems wrong, but I'd love another opinion for "Building Integrations".
- SAML is one of our enterprise-only features, so it seemed like we wouldn't want to mix that in with "regular" console documentation. (This was also how Heroku document their enterprise-only functionality.)
- I put in `aliases` for the pages to hopefully not break any links. Is that sufficient? Do we have any way to verify we don't break links deployment-to-deployment?

Part of https://github.com/pulumi/docs/issues/316.

### CODE REVIEW

File: content/docs/console/integrations/pulumi-button.md [hunk 1]
```diff
@@ -23,7 +25,8 @@ There are two steps to create a button:
 
 The Pulumi button works with template projects hosted in public GitHub repositories or Gists. A template is a Pulumi project that has the required `Pulumi.yaml` file, which describes the project. The template project can be in the root of the GitHub repository or within a subdirectory. Multiple projects can be hosted within subdirectories of a single repository.
 
-The `Pulumi.yaml` file can optionally contain a `template` section, which typically includes a `config` section used to specify any required config values for the project. Each config value can have a `description` and `default` value. Config values can also have a `secret` property, which can be set to `true` to indicate the config value is a [secret]({{< relref "../config.md#secrets" >}}).
+The `Pulumi.yaml` file can optionally contain a `template` section, which typically includes a `config` section used to specify any required config values for the project. Each config value can have a `description` and `default` value. Config values can also have a `secret` property, which can be set to `true` to indicate the config value is a
+[secret]({{< ref "/docs/reference/config.md#secrets" >}}).
```
Comment (justinvp) [inline content/docs/console/integrations/pulumi-button.md]: Why the change from `relref` to `ref`? (As an aside, I think we should do a pass throughout the site and change any `ref`s to just be `relref`, unless there is a need for the URL to be a fully qualified URL.

File: content/docs/console/integrations/webhooks.md [hunk 2]
```diff
@@ -55,6 +57,7 @@ Organization-level webhooks will be sent webhook events from all of the stacks w
 that organization.
 
 | Event Kind | Triggered |
+| --- | --- |
```
Comment (justinvp) [inline content/docs/console/integrations/webhooks.md]: I noticed the tables in the rest of this page aren't being rendered correctly, so perhaps a similar change is needed for the tables below.

File: content/docs/reference/commands.md [hunk 3]
```diff
@@ -9,7 +9,7 @@ menu:
 Pulumi is controlled primarily using the command line interface (CLI). It works in conjunction with the Pulumi service
 to deploy changes to your cloud apps and infrastructure.  It keeps a history of who updated what in your team and when.
 This CLI has been designed for great inner loop productivity, in addition to
-[continuous integration and deployment]({{< relref "cd.md" >}}) scenarios.
+[continuous integration and deployment]({{< ref "/docs/console/continuous-delivery" >}}) scenarios.
```
Comment (justinvp) [inline content/docs/reference/commands.md]: Nit: Existing nit but we tend to use "delivery" over "deployment".

```suggestion
[continuous integration and delivery]({{< ref "/docs/console/continuous-delivery" >}}) scenarios.
```

File: content/docs/reference/organizing-stacks-projects.md [hunk 4]
```diff
@@ -134,7 +134,8 @@ Pulumi project. However, most users find that a close alignment between Git repo
 structure enables seamless continuous deployment.
 
 In this model, there is a rough correspondence between a Git repo and a Pulumi project, and a Git branch and
-its associated Pulumi stack. Please read more about [how these mapping are maintained here]({{< relref "cd.md" >}}).
+its associated Pulumi stack. Please read more about
+[how these mapping are maintained here]({{< ref "/docs/console/continuous-delivery" >}}).
```
Comment (justinvp) [inline content/docs/reference/organizing-stacks-projects.md]:
```suggestion
[how these mapping are maintained here]({{< relref "/docs/console/continuous-delivery" >}}).
```

File: layouts/partials/docs/toc.html [hunk 5]
```diff
@@ -10,6 +10,11 @@ <h4 class="no-anchor">
         </h4>
         {{ template "toc" (dict "page" . "menu" .Site.Menus.quickstart) }}
 
+        <h4 class="no-anchor">
+            <a href="{{ relref . "/docs/console" }}">Pulumi Cloud Console</a>
```
[the remark this answers is not in the document]
Comment (bermudezmt) [reply layouts/partials/docs/toc.html]: Yeah, we may also want to make the fonts smaller on the left nav. And it took me a few days to realize you could click on the top-level nav items like "Get Started" and Reference" . I wonder if we should have a `caret-down` icon next to them? And either move the "index" content under an Overview page for the top-level item, or make it clearer that it's clickable and serves as the index or landing page for that category. @cnunciato - I'm sure you have thoughts around this, too.

File: content/docs/console/collaboration/_index.md [hunk 6]
```diff
@@ -0,0 +1,10 @@
+---
+title: "Collaboration"
```
[the remark this answers is not in the document]
Comment (bermudezmt) [reply content/docs/console/collaboration/_index.md]: So are the quotes not required? I do see them on a few pages.

[the remark this answers is not in the document]
Comment (chrsmith) [reply content/docs/console/collaboration/_index.md]: Done here, and on all the other pages.

File: layouts/partials/docs/toc.html (hunk 5 above)
[the remark this answers is not in the document]
Comment (chrsmith) [reply layouts/partials/docs/toc.html]: For the short-term, it may make the most sense to put it at the bottom. So no push back from me. But I think that the end-state should have Pulumi Cloud Console at the top and _maybe_ under "Get Started". 

The core issue is that "Pulumi" may mean two different things: the code-focused framework for creating cloud resources (e.g. CLI, code examples) or it is the SaaS application for tracking resource state (e.g. console, screenshots). Those are two very different things (and perhaps serving two different use-cases / situations).

I can see us eventually moving the Cloud Console docs to a separate section of the website entirely, perhaps splitting "product documentation" out from "library/package API reference". But like I said, we're probably still a ways away from that. So it's more important to just flesh out the content for the Cloud Console and put it _somewhere_.

We can find a better home later, when our L2 nav becomes even more crowded and we need to vote something off the island.

File: content/docs/console/continuous-delivery/_index.md [hunk 7]
```diff
@@ -1,9 +1,10 @@
 ---
```
[the remark this answers is not in the document]
Comment (chrsmith) [reply content/docs/console/continuous-delivery/_index.md]: Done.

File: content/docs/console/integrations/_index.md [hunk 8]
```diff
@@ -0,0 +1,10 @@
+---
+title: "Building Integrations"
```
[the remark this answers is not in the document]
Comment (chrsmith) [reply content/docs/console/integrations/_index.md]: Done. Though I think I'll move this to a "Building Extensions" section. (Along with a separate "Integrations" section for things like the GitHub App.)

File: content/docs/console/integrations/_index.md [hunk 9]
```diff
@@ -0,0 +1,10 @@
...
+
+redirect_to: /docs/console/integrations/webhooks/
```
[the remark this answers is not in the document]
Comment (chrsmith) [reply content/docs/console/integrations/_index.md]: SGTM

File: content/docs/console/integrations/pulumi-button.md (hunk 1 above)
Comment (chrsmith) [reply content/docs/console/integrations/pulumi-button.md]: Let's sync up about this, since if we kept them as `relref` it seems difficult to maintain if it becomes `"../../reference/config.md"`. Just having a qualified `ref` seems cleaner IMHO, though when we do a pass over all of our Service doc content, let's revisit this conversation.

File: content/docs/console/integrations/webhooks.md (hunk 2 above)
Comment (chrsmith) [reply content/docs/console/integrations/webhooks.md]: Yeah, thanks. Fixed.

File: layouts/partials/docs/toc.html (hunk 5 above)
[the remark this answers is not in the document]
Comment (cnunciato) [reply layouts/partials/docs/toc.html]: > or make it clearer that it's clickable

💯 yep &mdash; it's been on my mental list, but I haven't had a chance to propose an alternative yet. Will work on that.

@chrsmith This is great. I finally know where all this stuff is, now. 😄 🍹

**Comment by lukehoban** [comment]
When we have a section, we typically have content in the topic that is the section itself which gives an overview of the material for that topic. It seems like we’re missing those pages for several of the new sections?

Particularly because of that - it feels like we might not be ready for a Pulumi Enterprise section. If we have it - it has to describe what Pulumi Enterprise is. But I think the pricing page is fine on that for now.  Since it only currently has SSO - I suggest we put it under Collaboration but just have a “Pulumi Enterprise” note at the top of the page.

**Comment by chrsmith** [comment]
@lukehoban Yes, I agree. I'll have meaningful landing pages for each of those sections in the next PR which fills in the doc gaps for the service. Though you do raise a good point about perhaps not being ready for a dedicated section specific to the Enterprise offering. We'll see how the rest of the doc pages work out.
